src/ucs/datastruct/ptr_array.c

```diff
@@ -54,16 +49,39 @@ ucs_ptr_array_freelist_set_next(ucs_ptr_array_elem_t *elem, unsigned next)
                     (((ucs_ptr_array_elem_t)next) << UCS_PTR_ARRAY_NEXT_SHIFT);
 }
 
+static inline unsigned
+ucs_ptr_array_freelist_get_next_params(ucs_ptr_array_elem_t elem,
+                                       uint32_t *size_free)
+{
+    ucs_assert(__ucs_ptr_array_is_free(elem));
+    *size_free = elem >> UCS_PTR_ARRAY_SIZE_FREE_SHIFT;
+    return ((elem & UCS_PTR_ARRAY_NEXT_MASK) >> UCS_PTR_ARRAY_NEXT_SHIFT);
+}
+
+static inline void
+ucs_ptr_array_freelist_element_set_free(ucs_ptr_array_elem_t *elem,
+                                        uint32_t size_free,
+	unsigned next)
+{
+    ucs_assert(next <= UCS_PTR_ARRAY_NEXT_MASK);
+
+    *elem = UCS_PTR_ARRAY_FLAG_FREE |
+	(((ucs_ptr_array_elem_t)size_free) << UCS_PTR_ARRAY_SIZE_FREE_SHIFT) |
+	(((ucs_ptr_array_elem_t)next) << UCS_PTR_ARRAY_NEXT_SHIFT);
+}
+
 static void UCS_F_MAYBE_UNUSED ucs_ptr_array_dump(ucs_ptr_array_t *ptr_array)
 {
 #if UCS_ENABLE_ASSERT
     unsigned i;
 
-    ucs_trace_data("ptr_array start %p size %u", ptr_array->start, ptr_array->size);
+    ucs_trace_data("ptr_array start %p size %u",
+                   ptr_array->start, ptr_array->size);
```

Comment:
why move to the new line? it is longer than 80 symbols?

Reply:
Yes...